build_tools/python/cmake_builder/rules_test.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+## Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import unittest
+import cmake_builder.rules
+
+
+class RulesTest(unittest.TestCase):
+
+  def test_get_target_path(self):
+    self.assertEqual(cmake_builder.rules.get_target_path("abcd"),
+                     "${_PACKAGE_NAME}_abcd")
+
+  def test_build_iree_bytecode_module(self):
+    rule = cmake_builder.rules.build_iree_bytecode_module(
+        target_name="abcd",
+        src="abcd.mlir",
+        module_name="abcd.vmfb",
+        flags=["--backend=cpu", "--opt=3"],
+        compile_tool_target="iree_iree-compile2",
+        c_identifier="abcd.c",
+        static_lib_path="libx.a",
+        deps=["iree_libx", "iree_liby"],
+        testonly=True,
+        public=False)
+
+    self.assertEqual(rule, ('iree_bytecode_module(\n'
+                            '  NAME\n'
+                            '    "abcd"\n'
+                            '  SRC\n'
+                            '    "abcd.mlir"\n'
+                            '  MODULE_FILE_NAME\n'
+                            '    "abcd.vmfb"\n'
+                            '  C_IDENTIFIER\n'
+                            '    "abcd.c"\n'
+                            '  COMPILE_TOOL\n'
+                            '    "iree_iree-compile2"\n'
+                            '  STATIC_LIB_PATH\n'
+                            '    "libx.a"\n'
+                            '  FLAGS\n'
+                            '    "--backend=cpu"\n'
+                            '    "--opt=3"\n'
+                            '  DEPS\n'
+                            '    "iree_libx"\n'
+                            '    "iree_liby"\n'
+                            '  TESTONLY\n'
+                            ')\n'))
```

Comment:
This seems a bit hard to maintain. Maybe triple quotes and textwrap.dedent would be more readable here? 

```suggestion
    self.assertEqual(
        rule,
        textwrap.dedent("""\
        iree_bytecode_module(
          NAME
            "abcd"
          SRC
            "abcd.mlir"
          MODULE_FILE_NAME
            "abcd.vmfb"
          C_IDENTIFIER
            "abcd.c"
          COMPILE_TOOL
            "iree_iree-compile2"
          STATIC_LIB_PATH
            "libx.a"
          FLAGS
            "--backend=cpu"
            "--opt=3"
          DEPS
            "iree_libx"
            "iree_liby"
          TESTONLY
        )
        """))
```